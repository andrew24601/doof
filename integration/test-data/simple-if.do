// Test simple if-else control flow
// Tests: JMP_IF_FALSE, JMP_IF_TRUE, JMP

function main(): int {
    let x: int = 10;
    let y: int = 5;
    
    if (x > y) {              // LT_INT + JMP_IF_FALSE
        println("x is greater");
    } else {
        println("y is greater or equal");
    }
    
    if (x < y) {              // GT_INT + JMP_IF_FALSE  
        println("x is less");
    } else {
        println("x is greater or equal");
    }
    
    return 0;
}
