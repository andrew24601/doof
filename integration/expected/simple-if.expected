x is greater
x is greater or equal
